### PR TITLE
Jsontype

### PR DESCRIPTION
**Pull request by rdaly525**
Added a json type to Values.

### CODE REVIEW

**Comment by rdaly525** [comment]
@leonardt, could your review this when you have a chance?

File: include/coreir/ir/value.h [hunk 1]
```diff
@@ -99,22 +101,6 @@ class Arg : public Value {
...
-//    static std::shared_ptr<CPPLambda> make(LambdaType lambda) {
-//      return std::make_shared<CPPLambda>(lambda);
-//    } 
-//  private :
-//    LambdaType lambda;
-//  public :
-//    bool operator==(const Value& r) const {
-//      return false;
-//    }
-//};
```
Comment (leonardt) [inline include/coreir/ir/value.h]: Should this be removed? You removed `VK_CPPLambda` from the enum list, but you kept this around. Should either keep both commented out for consistency (so you don't forget it if you're going to add it back in), or just removed both of them.

File: include/coreir/ir/value.h [hunk 2]
```diff
@@ -206,9 +194,6 @@ class TemplatedConst : public Const {
     bool operator<(const Value& r) const override;
 
     static bool classof(const Value* v) {return v->getKind()==Underlying2Kind<T>::kind;}
-    //static std::shared_ptr<TemplatedConst<T>> make(ValueType* type, T value) {
-    //  return std::make_shared<TemplatedConst<T>>(type,value);
-    //}
```
Comment (leonardt) [inline include/coreir/ir/value.h]: Not sure why the two above methods are removed.

File: include/coreir/ir/valuecache.h [hunk 3]
```diff
@@ -10,16 +10,22 @@ class BitVectorComp {
     bool operator() (const BitVector& l, const BitVector& r) const;
 };
 
+//class JsonComp {
+//  public:
+//    bool operator() (const Json& l, const Json& r) const;
+//};
```
Comment (leonardt) [inline include/coreir/ir/valuecache.h]: Do you still need this?

File: src/ir/value.cpp [hunk 4]
```diff
@@ -103,30 +103,13 @@ Const* Const_impl<Module*>(Context* c,Module* val) {
...
-//
-//template<>
-//string TemplatedConst<std::string>::toString() const {
-//  return vtype->toString() + "(" + value + ")";
-//}
-//
-//template<>
-//string TemplatedConst<Type*>::toString() const {
-//  return vtype->toString() + "(" + value->toString() + ")";
-//}
```
Comment (leonardt) [inline src/ir/value.cpp]: All these toString methods are being removed?

**Comment by rdaly525** [comment]
I do not have plans on using any other json library other than nholmann's.

File: include/coreir/ir/value.h (hunk 2 above)
Comment (rdaly525) [reply include/coreir/ir/value.h]: This was never in the code base. It was always commented. Probably should have removed it in a different PR

File: src/ir/value.cpp (hunk 4 above)
Comment (rdaly525) [reply src/ir/value.cpp]: Similar to the other comment, these were being unused. there was already a toString method defined for this class.

**Pull request merged**